api/v1alpha1/hostedcluster_types.go

```diff
@@ -295,104 +383,146 @@ const (
 type AWSEndpointAccessType string
 
 const (
-	// Public endpoint access allows public kube-apiserver access and public node communication with the control plane
+	// Public endpoint access allows public API server access and public node
+	// communication with the control plane.
 	Public AWSEndpointAccessType = "Public"
 
-	// PublicAndPrivate endpoint access allows public kube-apiserver access and private node communication with the control plane
+	// PublicAndPrivate endpoint access allows public API server access and
+	// private node communication with the control plane.
 	PublicAndPrivate AWSEndpointAccessType = "PublicAndPrivate"
 
-	// Private endpoint access allows only private kube-apiserver access and private node communication with the control plane
+	// Private endpoint access allows only private API server access and private
+	// node communication with the control plane.
 	Private AWSEndpointAccessType = "Private"
 )
 
+// AWSPlatformSpec specifies configuration for clusters running on Amazon Web Services.
 type AWSPlatformSpec struct {
-	// Region is the AWS region for the cluster.
-	// This is used by CRs that are consumed by OCP Operators.
-	// E.g cluster-infrastructure-02-config.yaml and install-config.yaml
-	// This is also used by nodePools to fetch the default boot AMI in a given payload.
+	// Region is the AWS region in which the cluster resides. This configures the
+	// OCP control plane cloud integrations, and is used by NodePool to resolve
+	// the correct boot AMI for a given release.
+	//
 	// +immutable
 	Region string `json:"region"`
 
-	// CloudProviderConfig is used to generate the ConfigMap with the cloud config consumed
-	// by the Control Plane components.
+	// CloudProviderConfig specifies AWS networking configuration for the control
+	// plane.
+	//
+	// TODO(dan): should this be named AWSNetworkConfig?
+	//
 	// +optional
 	// +immutable
 	CloudProviderConfig *AWSCloudProviderConfig `json:"cloudProviderConfig,omitempty"`
 
-	// ServiceEndpoints list contains custom endpoints which will override default
-	// service endpoint of AWS Services.
-	// There must be only one ServiceEndpoint for a service.
+	// ServiceEndpoints specifies optional custom endpoints which will override
+	// the default service endpoint of specific AWS Services.
+	//
+	// There must be only one ServiceEndpoint for a given service name.
+	//
 	// +optional
 	// +immutable
 	ServiceEndpoints []AWSServiceEndpoint `json:"serviceEndpoints,omitempty"`
 
+	// Roles must contain exactly 3 entries representing the locators for roles
+	// supporting the following OCP services:
+	//
+	// - openshift-ingress-operator/cloud-credentials
+	// - openshift-image-registry/installer-cloud-credentials
+	//  -openshift-cluster-csi-drivers/ebs-cloud-credentials
+	//
+	// Each role has unique permission requirements whose documentation is TBD.
+	//
+	// TODO(dan): revisit this field; it's really 3 required fields with specific content requirements
+	//
```

Comment:
FWIW I'd be +1 to be more rigid here and move this into specific list of fields. Probably to discuss in different PR.